src/Ppu.re

```diff
@@ -0,0 +1,164 @@
+type registers = {
+  mutable control: int,
+  mutable mask: int,
+  mutable status: int,
+  mutable oam_address: int,
+  mutable oam_data: int,
+  mutable ppu_address: int,
+  mutable ppu_data: int,
+  mutable coarse_x: int,
+  mutable coarse_y: int,
+  mutable fine_x: int,
+  mutable fine_y: int,
+  mutable buffer: int,
+  mutable write_latch: bool
+};
+
+type vblank_nmi =
+  | NMIEnabled
+  | NMIDisabled;
+
+type t = {
+  registers: registers,
+  oam: array(int),
+  name_table: array(int),
+  palette_table: array(int),
+  pattern_table: Rom.t
+};
+
+let build = (rom) => {
+  {
+    registers: {
+      control: 0,
+      mask: 0,
+      status: 0,
+      oam_address: 0,
+      oam_data: 0,
+      ppu_address: 0,
+      ppu_data: 0,
+      coarse_x: 0,
+      coarse_y: 0,
+      fine_x: 0,
+      fine_y: 0,
+      buffer: 0,
+      write_latch: false
+    },
+    oam: Array.make(0x100, 0),
+    name_table: Array.make(0x800, 0),
+    palette_table: Array.make(0x20, 0),
+    pattern_table: rom
+  };
+};
+
+let ctrl_helper = (n, unset, set) => {
```

Comment:
Couldn't this be `let ctrl_helper = (n, unset, set, regs) => ...`?

Reply:
Ah, the partial application trick! Yes, thank you. :)